cli/build.sbt

```diff
@@ -1,11 +1,18 @@
 import sbt._
-
+import scala.collection.immutable.Seq
 
 // Disable GC since the CLI is a short-lived process.
 nativeGC := "none"
 
-nativeLinkingOptions := Seq(
-  "-lcurl",
-  "-L", (baseDirectory.value / ".." / "libhttpsimple" / "target").toPath.toAbsolutePath.toString
-) ++ sys.props.get("nativeLinkingOptions").fold(Seq.empty[String])(_.split(" "))
+nativeLinkingOptions := {
+  val dynamicLinkerOptions =
+    Properties
+      .dynamicLinker
+      .toVector
+      .map(dl => s"-Wl,--dynamic-linker=$dl")
```

Comment:
This is used by the musl-based builds where we're packaging our own dynamic linker with the application. This is done to support old environments, e.g. Ubuntu 14.04 and CentOS6 which are both still supported and may be in use at ICP environments.